enlighten/device/BatteryFeature.py

```diff
@@ -69,94 +48,95 @@ def process_reading(self, spec, reading):
 
         app_state = spec.app_state
         rds = app_state.battery_data
-        self.raw = reading.battery_raw
-        self.perc = reading.battery_percentage
-        self.charging = reading.battery_charging
-        rds.add(self.perc)
+
+        raw = reading.battery_raw
+        perc = reading.battery_percentage
+        is_charging = reading.battery_charging
+        charging_label = 'charging' if is_charging else 'discharging'
+
+        log.debug(f"adding {perc} to RDS")
+        rds.add(perc)
         current_time = datetime.datetime.now()
         if self.output_to_file:
-            self.hardware_file_manager.write_line(self.name,f"{self.name},{spec.label},{current_time}, {self.perc}, charging: {self.charging}")
+            self.ctl.hardware_file_manager.write_line(self.name,f"{self.name},{spec.label},{current_time}, {perc}, {charging_label}")
 
-        self._update_labels(reading)
+        self.ctl.form.ui.label_battery_raw.setText("0x%06x" % reading.battery_raw)
+        self.ctl.form.ui.label_battery_parsed.setText(f"Battery ({perc:.2f}%, {charging_label})")
+        if spec == current_spec:
+            self.lb_perc.setText(f"{perc:.2f} %")
+            
         for cb in self.observers:
-            cb(self.perc, self.charging)
+            cb(perc, is_charging)
 
-        active_curve = self.multispec.get_hardware_feature_curve(self.name, spec.device_id)
-        if active_curve == None:
+        # update graph on Factory View
+        active_curve = self.ctl.multispec.get_hardware_feature_curve(self.name, spec.device_id)
+        if active_curve is None:
             return
-
-        x_time = [(current_time-x).total_seconds() for x,y in rds.data]
-        y = rds.get_values()
-
         try:
-            self.graph.set_data(curve=active_curve, y=y, x=x_time)
+            x, y = rds.get_relative_to_now()
```

Comment:
encapsulated within RollingDataSet so we can easily add more time-based Factory charts